### ISSUE TITLE
Claims PIN is bad. Restore from backup? User Documentation?

### ISSUE THREAD

**Issue by smscotten** [voisine/breadwallet#154]
I tried to send an amount of BTC tonight, transfer *in* to Breadwallet went without a hitch. When spending, however, Breadwallet is not accepting the PIN. I wrote the PIN down at the time that I set it, so I know that it is correct.

I have the Recovery Phrase, and I assume that it's impossible for that to get arbitrarily reset. But I don't want to "wipe" my wallet without having some idea of what that means. Also, if I recover the wallet, does that let me reset the PIN without the old PIN?

If I'm completely screwed on this, how is it possible that my PIN could have evaporated? There was something in the other issues about the PIN being stored in the Apple keychain. Well, in order to try to address some unrelated issues, I turned off the iCloud keychain and turned it back on again. If Breadwallet is checking against a PIN that is no longer present on my device am I out of luck even if I have the right PIN (which I still absolutely believe that I do).

Basically: I have the recovery phrase. Can I reset the PIN?

But also: this seems like a real end-user question. I hate to come whining on github where I really should have RTFMed first. I've been looking for some form of documentation but it looks like this is it. Anything I missed?

**Comment by voisine** [comment]
We're doing support through github for now, so no worries.

If you have your recovery phrase, you'll be fine. After three failed passcode attempts you will get locked out for a time, with the option to reset the passcode using your recovery phrase.

It is strange that passcode you wrote down isn't working. Maybe you changed it at some point and forgot?

**Comment by smscotten** [comment]
Thanks. I have satisfied myself that the BTC is safe/accessible. I used another device to restore, set a new pin on the new device and spent away. So I'm good. I didn't want to do much more on the phone to avoid making things worse.

My guess is that the passcode issue has something to do with me disconnecting from iCloud keychain and reconnecting. Does that make any sense? I haven't looked at how Keychain works at all.

I'm quite certain I didn't change the PIN again. It was only a few days ago that I installed Breadwallet. If it was just a matter of my imperfect memory I'd entertain the likelihood that I'd just space-cadeted it. But I put the PIN into my password manager on the laptop at the same time that I set it—within seconds. I'm not going to pretend I couldn't forget a PIN but that's exactly why I don't try to remember stuff like that.

**Comment by voisine** [comment]
Fair enough. I will mess around with iCloud keychain settings and see if I can get anything bad to happen.

**Comment by Hegetiby** [comment]
Hi, I guess I am in a mess, but either ways asking for some info here trying to recover my bitcoins. Had 1.6 BTC on my wallet and accidentally restored content to fabric on my Iphone, so, all the info got erased, the worst part is I lost the paper with the phrase and didn't memorized it :( I am just counting with the Itunes backup,is there any way of accessing my BTC? Restored my phone but seems like it's only new wallet or restore with the phrase. Thanks!

**Comment by voisine** [comment]
@Hegitiby do you mean you selected "erase all content and settings" on the phone?

If you have iTunes or iCloud backup, and the same phone the backup was made with, in theory you should be able to get your wallet back. I would suggest turning off iCloud backup on the phone to make sure it doesn't get overwritten.

I haven't actually tried to restore a wallet from iTunes backup to the same phone. I will see if I can do it. Which do you have by the way, iCloud or iTunes backup? And you know the backup was made after you created your wallet but before you erased the phone?

**Comment by Hegetiby** [comment]
@voisine Yes,erased all,had the wallet on an iphone 4 which I borrowed to a friend but erased all content first. Afterwards I realized I didn't had Bread on the Iphone 5S because it was jailbroken. Stopped the iCloud backup for my phone not to erase or overwrite the other one in some way since they use the same account but looks like ICloud had different backups for every phone. Will try restoring that phone to the Icloud backup before deleting all and let you know, that would be awesome. Just need to get it back. Thanks for the answer! :) You gave me a ray of hope

**Comment by Hegetiby** [comment]
BTW,I will use Bread now even more, this just tells me I under estimated it a bit maybe by not giving attention to the phrase that much. You could have a full red screen before installing to warn or scare people of the consequences, haha know you did, just thinking of people like me. :p

**Comment by voisine** [comment]
@Hegetiby that's great, please let me know if it works, I believe it should. The breadwallet private key is set to "this device only" so that an attacker can't steal your bitcoins by hacking into your iCloud account. They would need physical access to your phone and the ability to unlock it.

I was thinking about this problem, people not writing down their recovery phrase is going to be by far the biggest cause of loss. The only secure options for backup of private keys though is paper, or a hardened, locked down device with a passcode like an iPhone. Since lots of people have multiple devices, I'm thinking there could be an option to send a wallet backup to another iOS device incase yours gets lost.

**Comment by Hegetiby** [comment]
Good news!! Got my Bitcoins back! You are a genious! Apple didn't wanted me to, since they wouldn't let me restore the latest backup asking me ios 8.1.2, having an Iphone 4 with 7.1.2 as the last possible one. Don't understand,  just beacause I shared the Icloud account with an iphone 5s they overwrote the 4 backup to the 5s's. Thank God I had a backup I kept out of the Backup folder on my external disk on my computer and could restore to that one and got my wallet back. I wrote the phrase on a paper end kept it well now. :D

**Comment by voisine** [comment]
Excellent, I'm really glad you didn't lose funds. I've made it my mission to help people safely use bitcoin and reduce loss to the absolute minimum possible, both by data security and by design of the user experience.

Your report got me thinking much harder about how to implement automatic and safe backups, hopefully every single user, since expecting people to write down their recovery phrase is too much to ask.

**Comment by Hegetiby** [comment]
Great work :D I am back to BreadWallet and more in love with it than ever before. This experience made me see how secure this wallet is and take more responsibility on my side.

**Comment by Hegetiby** [comment]
A quick question,why does it take so much time to send a payment, it give me error,payment cannot be made, network timeout, tried on 3g and wifi, the same.

**Comment by voisine** [comment]
@Hegetiby, did it just take a long time, or did it fail? Did it just happen one time, or is it every time? Did you see "not connected" on the send screen?

**Comment by Hegetiby** [comment]
Updated the app,did the same, but after connecting, disconnecting internet, initiated the transaction but failed to finish it and asked me to sync again, just did it and now I see the transaction is getting confirmations, so guess it works :)

**Comment by voisine** [comment]
I'll see if I can improve how it handles sending when not connected. It may ask you to sync in some situation when it doesn't need to because it was just an intermittent connection issue.

**Comment by mporter15** [comment]
I lost my breadwallet passcode but I have the recovery phrase. Can someone tell me the steps to reset the passcode?

**Comment by voisine** [comment]
@mporter15 there's a UI bug in the current release, so there's two options.

You can send your iTunes account email address to support@breadwallet.com and we can invite you to the beta program for an update with the fix, or you alternately you can keep entering different wrong passcodes until the wallet resets after the 8th attempt (and about 24hrs of being disabled) and then select "recover wallet"

**Comment by serg720** [comment]
I too have an issue. I got locked out of my pin and can't remember my phrase. Had money in there, did a recover/start new one and my money is gone. any suggestions??? thank you

**Comment by voisine** [comment]
@serg720 please contact support@breadwallet.com if you're having trouble

when you "start/recover another wallet" that app makes you type in the recovery phrase of the existing wallet to prove you wrote it down

**Comment by serg720** [comment]
Hi Arron thanks for the reply. I've tried everything. Have two separate recovery codes and it takes me to the account where there is no money.

Sent from my iPhone

On Nov 8, 2016, at 2:59 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


@serg720<https://github.com/serg720> please contact support@breadwallet.com<mailto:support@breadwallet.com> if you're having trouble

when you "start/recover another wallet" that app makes you type in the recovery phrase of the existing wallet to prove you wrote it down

-

**Comment by voisine** [comment]
if you see no transactions listed after doing a wallet recover, you can verify that blockchain scan worked correctly (and that the wallet for that phrase is indeed empty) by doing a lookup of the receive address on a block explorer

**Comment by serg720** [comment]
Yeah I recovered but didn't see my transaction. How do I do the block explorer?

Sent from my iPhone

On Nov 8, 2016, at 4:04 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


if you see no transactions listed after doing a wallet recover, you can verify that blockchain scan worked correctly (and that the wallet for that phrase is indeed empty) by doing a lookup of the receive address on a block explorer

-

**Comment by voisine** [comment]
copy your receive address to the clipboard and paste it into the search form of any block explorer site
type "block explorer" into google

**Comment by serg720** [comment]
Ok just did that and it's showing a transaction, a few of them. Now what?

Sent from my iPhone

On Nov 8, 2016, at 4:19 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


copy your receive address to the clipboard and paste it into the search form of any block explorer site
type "block explorer" into google

-

**Comment by serg720** [comment]
It's showing a 27.00 transaction but there should be a 180.00 from bovada to bread wallet

Sent from my iPhone

On Nov 8, 2016, at 4:19 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


copy your receive address to the clipboard and paste it into the search form of any block explorer site
type "block explorer" into google

-

**Comment by voisine** [comment]
try "rescan blockchain" in settings if the app is missing transactions

**Comment by serg720** [comment]
Did the rescan a few times yesterday and nothing. I have my old address : 19R51Va84dmukgPoSXudVgr3EqrUqV7iPP. This address is from the one I should've gotten the deposit in



Sent from my iPhone

On Nov 8, 2016, at 4:46 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


try "rescan blockchain" in settings if the app is missing transactions

-

**Comment by voisine** [comment]
so you don't see the transactions listed in the block explorer in your breadwallet with the same receive address?

**Comment by serg720** [comment]
Nope. It shows some 30.00 and a 27.00 transaction but not the one when bovada paid out for 180.00 on nov 3rd[image1.PNG]



Sent from my iPhone

On Nov 8, 2016, at 4:59 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


so you don't see the transactions listed in the block explorer in your breadwallet with the same receive address?

-

**Comment by voisine** [comment]
what's the transaction ID for the one that's missing?

**Comment by serg720** [comment]
From bovada? I'm sorry to bug you. If we can chat so I can tell you in detail 619-279-0465 that would be great

Sent from my iPhone

On Nov 8, 2016, at 5:02 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


what's the transaction ID for the one that's missing?

-

**Comment by voisine** [comment]
If you need app support, we provide that through email, support@breadwallet.com

The GitHub issue tracker is for bugs and feature requests, which is publicly viewable.

**Comment by serg720** [comment]
They stop replying. I mean if I was able to bring up those old transactions, how come the 180.00 one is missing???

Sent from my iPhone

On Nov 8, 2016, at 5:09 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


If you need app support, we provide that through email, support@breadwallet.com<mailto:support@breadwallet.com>

The GitHub issue tracker is for bugs and feature requests, which is publicly viewable.

-

**Comment by voisine** [comment]
try pinging the support email again

if one is missing but the others show up, my guess is the missing transaction either wasn't sent or it was sent to a different receive address... you can find out by getting the bitcoin transaction id from the sender and looking it up on a block explorer

**Comment by serg720** [comment]
Ok I'll keep trying. Do iPhones have ip addresses? I see it's an option too

Sent from my iPhone

On Nov 8, 2016, at 5:18 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


try pinging the support email again

if one is missing but the others show up, my guess is the missing transaction either wasn't sent or it was sent to a different receive address... you can find out by getting the bitcoin transaction id from the sender and looking it up on a block explorer

-

**Comment by voisine** [comment]
any device that connects to the internet has an ip address

ip addresses have nothing to do with bitcoin addresses

**Comment by serg720** [comment]
Ok I'll call bovada and see if they have record of that. I saw it go into my bread wallet then I couldn't remember my pin and locked me out. So strange. Frustrating

Sent from my iPhone

On Nov 8, 2016, at 5:22 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


any device that connects to the internet has an ip address

ip addresses have nothing to do with bitcoin addresses

-

**Comment by voisine** [comment]
I'm glad you recovered the wallet you were using with your recovery phrase. If you used that same wallet for all your transactions, then you should be good. Most likely anything missing is a mistake by the sender.

**Comment by serg720** [comment]
Yeah I saved that part I just don't remember we any phrases or setting up a pin.

Sent from my iPhone

On Nov 8, 2016, at 5:32 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm glad you recovered the wallet you were using with your recovery phrase. If you used that same wallet for all your transactions, then you should be good. Most likely anything missing is a mistake by the sender.

-

**Comment by serg720** [comment]
Ok within 24-48 hrs they are going to send me that bitcoin address I need. Thanks

Sent from my iPhone

On Nov 8, 2016, at 5:32 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm glad you recovered the wallet you were using with your recovery phrase. If you used that same wallet for all your transactions, then you should be good. Most likely anything missing is a mistake by the sender.

-

**Comment by serg720** [comment]
Hi Aaron. I finally got that BTC number. I put it on that block chain w baited and there it is. What do I do next?

Sent from my iPhone

On Nov 8, 2016, at 5:32 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm glad you recovered the wallet you were using with your recovery phrase. If you used that same wallet for all your transactions, then you should be good. Most likely anything missing is a mistake by the sender.

-

**Comment by serg720** [comment]
Hey Aaron I have the BTC number now. What do I do?

Sent from my iPhone

On Nov 8, 2016, at 5:32 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm glad you recovered the wallet you were using with your recovery phrase. If you used that same wallet for all your transactions, then you should be good. Most likely anything missing is a mistake by the sender.

-

**Comment by serg720** [comment]
Hey Aaron I have the BTC number now. What do I do?

Sent from my iPhone

On Nov 10, 2016, at 10:59 PM, Sergio Navarro <serg720@hotmail.com<mailto:serg720@hotmail.com>> wrote:

Hey Aaron I have the BTC number now. What do I do?

**Comment by voisine** [comment]
@serg720 GitHub/Facebook/twitter/app store reviews/personal email addresses/etc are not for support requests

Use your recovery phrase to recover your wallet. If your funds aren't there, try the other one. If it's still not there, then it's probably on some other wallet you created and didn't write down the phrase to, or you somehow gave the sender an incorrect bitcoin address. If you lose your physical wallet with cash in it, you don't pester the company that manufactured the wallet. We are not a bank. We don't have your money.

**Comment by serg720** [comment]
You told me to recover the BTC address and I did. I scanned it and it won't show up in my breadwallet account.

Sent from my iPhone

On Nov 11, 2016, at 2:12 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


@serg720<https://github.com/serg720> GitHub/Facebook/twitter/app store reviews/personal email addresses/etc are not for support requests

Use your recovery phrase to recover your wallet. If your funds aren't there, try the other one. If it's still not there, then it's probably on some other wallet you created and didn't write down the phrase to, or you somehow gave the sender an incorrect bitcoin address. If you lose your physical wallet with cash in it, you don't pester the company that manufactured the wallet. We are not a bank. We don't have your money.

-

**Comment by voisine** [comment]
this is not the correct for support requests

**Comment by serg720** [comment]
And your customer service department has been ignoring me. I need my money

Sent from my iPhone

On Nov 11, 2016, at 2:12 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


@serg720<https://github.com/serg720> GitHub/Facebook/twitter/app store reviews/personal email addresses/etc are not for support requests

Use your recovery phrase to recover your wallet. If your funds aren't there, try the other one. If it's still not there, then it's probably on some other wallet you created and didn't write down the phrase to, or you somehow gave the sender an incorrect bitcoin address. If you lose your physical wallet with cash in it, you don't pester the company that manufactured the wallet. We are not a bank. We don't have your money.

-

**Comment by voisine** [comment]
I looked into it... they've responded to dozens of your requests attempting to help. It looks like you probably lost the recovery phrase for the wallet you used, or you mistakenly gave the sender the wrong bitcoin address somehow.

**Comment by serg720** [comment]
What sender? I was cashing out and they sent me the right confirmation of the transaction and the it's not showing on the app. From there I didn't send money to anyone.

Sent from my iPhone

On Nov 11, 2016, at 2:19 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


I looked into it... they've responded to dozens of your requests attempting to help. It looks like you probably lost the recovery phrase for the wallet you used, or you mistakenly gave the sender the wrong bitcoin address somehow.

-

**Comment by serg720** [comment]
Well then please tell your "costumer service" dept to help me solve this. It's my money that I need

Sent from my iPhone

On Nov 11, 2016, at 2:22 PM, Aaron Voisine <notifications@github.com<mailto:notifications@github.com>> wrote:


this is not the correct forum for support requests

-

**Comment by samuraisam** [comment]
This is not the correct forum for support requests. Please continue your correspondence with support@breadwallet.com.
